### PR TITLE
Pxapi2/feature/pbi 158632 selection expressions

### PR DESCRIPTION
**Pull request by MikaelNordberg**
Support for selection expressions containing *, ? and TOP

### CODE REVIEW

File: PxWeb/Code/Api2/DataSelection/SelectionHandler.cs [hunk 1]
```diff
@@ -115,27 +245,217 @@ private VariablesSelection AddVariables(VariablesSelection variablesSelection, P
...
                 {
-                    var valueList = variable.ValueCodes[0].ToString().Split(',').ToList();
-                    selection.ValueCodes.AddRange(valueList.ToArray());
+                    AddWildcardStarValues(variable, values, value);
+                }
+                else if (value.Contains('?'))
+                {
+                    AddWildcardQuestionmarkValues(variable, values, value);
+                }
+                else if (value.Contains("TOP("))
```
Comment (likp) [inline PxWeb/Code/Api2/DataSelection/SelectionHandler.cs:282]: Same case sensitivity and starts with comment as above

**Review by likp — COMMENTED** [review]
See the comments for makeing the TOP expression case insensitive

File: PxWeb/Code/Api2/DataSelection/SelectionHandler.cs [hunk 2]
```diff
@@ -54,36 +71,149 @@ public bool Verify(PXModel model, VariablesSelection? variablesSelection, out Pr
...
+        }
+
+        /// <summary>
+        /// Returns true if the value string is a selection expression, else false.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        private bool IsSelectionExpression(string value)
+        {
+            return value.Contains('*') || value.Contains('?') || value.Contains("TOP(");
```
[the remark this answers is not in the document]
Comment (MikaelNordberg) [reply PxWeb/Code/Api2/DataSelection/SelectionHandler.cs]: Now checks that it starts with TOP and is case insensitive.

File: PxWeb/Code/Api2/DataSelection/SelectionHandler.cs (hunk 1 above)
Comment (MikaelNordberg) [reply PxWeb/Code/Api2/DataSelection/SelectionHandler.cs]: Now checks that it starts with TOP and is case insensitive.

**Pull request merged**